frontend/src/components/LoxotronCasinoGameComponent/LoxotronCasinoSlotMachine/index.tsx

```diff
@@ -0,0 +1,130 @@
+import { useRef, useEffect } from 'react';
+import frameUrl from '../../../assets/svg/Frame.svg';
+import './slotMachine.sass'
+import svg1 from '../../../assets/svg/tiles/1.svg';
+import svg2 from '../../../assets/svg/tiles/2.svg';
+import svg3 from '../../../assets/svg/tiles/3.svg';
+import svg4 from '../../../assets/svg/tiles/4.svg';
+import svg5 from '../../../assets/svg/tiles/5.svg';
+import svg6 from '../../../assets/svg/tiles/6.svg';
+import svg7 from '../../../assets/svg/tiles/7.svg';
+import svg8 from '../../../assets/svg/tiles/8.svg';
+import svg9 from '../../../assets/svg/tiles/9.svg';
+import svg10 from '../../../assets/svg/tiles/10.svg';
+
+
+const SlotMachineComponent = () => {
+    const canvasRef = useRef<HTMLCanvasElement>(null);
+
+    useEffect(() => {
+        if (!canvasRef.current) return;
+
+        const canvas = canvasRef.current as HTMLCanvasElement;
+        const context = canvas.getContext("2d");
+        const screenWidth = 1572;
+        const screenHeight = 622;
+
+        canvas.width = screenWidth;
+        canvas.height = screenHeight;
+
+        abstract class DrawableObject {
+            abstract draw(x: number, y: number): void;
+        }
+
+        class FillObject extends DrawableObject {
+            protected width: number = 1480;
+            protected height: number = 11;
+            protected x: number = 46;
+
+            draw(y: number) {
+                if(context) {
+                    context.fillStyle = "black";
+                    context.fillRect(this.x, y, this.width, this.height);
+                }
+            }
+        }
+
+        class SvgDrawableObject extends DrawableObject {
+            protected imageSrc: string;
+            protected x: number;
+            protected y: number;
+            constructor(x: number, y: number, imageSrc: string) {
+                super();
+                this.imageSrc = imageSrc;
+                this.x = x;
+                this.y = y;
+            }
+            draw() {
+                if(context) {
+                    const img = new Image();
```

Comment:
No need to create image each time you want to draw it. Create it in constructor and then left here context.drawImage which you will give this.img to draw the image